File: app/src/main/java/com/nvdevelopers/shoppingcart/activity/MainActivity.java
package com.nvdevelopers.shoppingcart.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.bottomnavigation.LabelVisibilityMode;
import com.nvdevelopers.shoppingcart.R;
import com.nvdevelopers.shoppingcart.fragments.AllCategory;
import com.nvdevelopers.shoppingcart.fragments.Cart;
import com.nvdevelopers.shoppingcart.fragments.Home;
import com.nvdevelopers.shoppingcart.fragments.MyAccount;

public class MainActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener {

    BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        loadFragment(new Home());

        bottomNavigationView = findViewById(R.id.bottom_navigation);
        bottomNavigationView.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_LABELED);
        bottomNavigationView.setOnNavigationItemSelectedListener(this);
    }

    private boolean loadFragment(Fragment fragment) {
        if (fragment != null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.container, fragment)
                    .commit();
            return true;
        }
        return false;
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        Fragment fragment = null;
        switch (item.getItemId()) {
            case R.id.bottom_navigation_menu_home:
                fragment = new Home();
                break;
            case R.id.bottom_navigation_menu_all_category:
                fragment = new AllCategory();
                break;
            case R.id.bottom_navigation_menu_cart:
                fragment = new Cart();
                break;
            case  R.id.bottom_navigation_menu_my_account:
                fragment = new MyAccount();
                break;
        }

        return loadFragment(fragment);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/nvdevelopers/shoppingcart/activity/MainActivity.kt
package com.nvdevelopers.shoppingcart.activity

import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.bottomnavigation.BottomNavigationView
import android.os.Bundle
import android.view.MenuItem
import androidx.fragment.app.Fragment
import com.nvdevelopers.shoppingcart.R
import com.nvdevelopers.shoppingcart.fragments.Home
import com.google.android.material.bottomnavigation.LabelVisibilityMode
import com.nvdevelopers.shoppingcart.fragments.AllCategory
import com.nvdevelopers.shoppingcart.fragments.Cart
import com.nvdevelopers.shoppingcart.fragments.MyAccount

class MainActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    var bottomNavigationView: BottomNavigationView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loadFragment(Home())
        bottomNavigationView = findViewById(R.id.bottom_navigation)
        bottomNavigationView?.setLabelVisibilityMode(LabelVisibilityMode.LABEL_VISIBILITY_LABELED)
        bottomNavigationView?.setOnNavigationItemSelectedListener(this)
    }

    private fun loadFragment(fragment: Fragment?): Boolean {
        if (fragment != null) {
            supportFragmentManager
                    .beginTransaction()
                    .replace(R.id.container, fragment)
                    .commit()
            return true
        }
        return false
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        var fragment: Fragment? = null
        when (item.itemId) {
            R.id.bottom_navigation_menu_home -> fragment = Home()
            R.id.bottom_navigation_menu_all_category -> fragment = AllCategory()
            R.id.bottom_navigation_menu_cart -> fragment = Cart()
            R.id.bottom_navigation_menu_my_account -> fragment = MyAccount()
        }
        return loadFragment(fragment)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return super.onOptionsItemSelected(item)
    }
}